modules/custom/social_language/src/Plugin/UserExportPlugin/UserLanguage.php

```diff
@@ -0,0 +1,96 @@
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\social_user_export\Plugin\UserExportPluginBase;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'UserLanguage' user export row.
+ *
+ * @UserExportPlugin(
+ *  id = "user_language",
+ *  label = @Translation("Language"),
+ *  weight = -340,
+ * )
+ */
+class UserLanguage extends UserExportPluginBase {
+
+  /**
+   * The language manager service.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  public $languageManager;
+
+  /**
+   * UserExportPluginBase constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Datetime\DateFormatterInterface $date_formatter
+   *   The date formatter.
+   * @param \Drupal\Core\Database\Connection $database
+   *   The database connection.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The group helper service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, DateFormatterInterface $date_formatter, Connection $database, LanguageManagerInterface $language_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $date_formatter, $database);
+
+    $this->languageManager = $language_manager;
+  }
+
+  /**
+   * The create method.
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   Container interface.
+   * @param array $configuration
+   *   An array of configuration.
+   * @param string $plugin_id
+   *   The plugin id.
+   * @param mixed $plugin_definition
+   *   The plugin definition.
+   *
+   * @return \Drupal\Core\Plugin\ContainerFactoryPluginInterface|\Drupal\social_user_export\Plugin\UserExportPluginBase
+   *   Returns the UserExportPluginBase.
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('date.formatter'),
+      $container->get('database'),
+      $container->get('language_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getHeader() {
+    return $this->t('Language');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getValue(UserInterface $entity):string {
+    return $this->languageManager->getLanguage($entity->getPreferredLangcode())->getName();
```

Comment:
Should we output the langcode instead? This may be slightly less friendly to human consumers but makes automation massively easier.

Reply:
I thought about this, but the language code itself is not very readable. I'll add another column for the language code.